### ISSUE TITLE
Input file is missing or of an unsupported image format

### ISSUE THREAD

**Issue by thitemple** [gatsbyjs/gatsby#6491]
## Summary

I just started developing a new blog and I have the following issue:

![image](https://user-images.githubusercontent.com/368734/42796342-8598c16e-8957-11e8-90de-f11e80011b7a.png)

I have no idea how to debug? In dev mode is working, this only happens when I run build. Is there a way to know which file is it talking about? I tried running with `--verbose` but that doesn't help.

## Relevant information

I'm running the latest beta from v2

### Environment (if relevant)

Windows 10

### File contents (if changed)

`gatsby-config.js`: 
```const config = require("./config/SiteConfig");

const pathPrefix = config.pathPrefix === "/" ? "" : config.pathPrefix;

module.exports = {
    pathPrefix: config.pathPrefix,
    siteMetadata: {
        title: "Blog",
        siteUrl: config.siteUrl + pathPrefix,
        buildTime: new Date(Date.now()).toLocaleString()
    },
    plugins: [
        "gatsby-plugin-feed",
        "gatsby-plugin-react-helmet",
        "gatsby-plugin-styled-components",
        {
            resolve: "gatsby-source-filesystem",
            options: {
                name: "post",
                path: `${__dirname}/blog`
            }
        },
        {
            resolve: "gatsby-transformer-remark",
            options: {
                plugins: [
                    {
                        resolve: "gatsby-remark-external-links",
                        options: {
                            target: "_blank",
                            rel: "nofollow noopener noreferrer"
                        }
                    },
                    "gatsby-remark-prismjs",
                    "gatsby-remark-autolink-headers",
                    {
                        resolve: `gatsby-remark-responsive-image`,
                        options: {
                            // It's important to specify the maxWidth (in pixels) of
                            // the content container as this plugin uses this as the
                            // base for generating different widths of each image.
                            maxWidth: 590
                        }
                    }
                ]
            }
        },
        {
            resolve: "gatsby-plugin-typography",
            options: {
                pathToConfigModule: "src/utils/typography.js"
            }
        },
        "gatsby-plugin-catch-links",
        "gatsby-plugin-sitemap",
        "gatsby-plugin-lodash",
        {
            resolve: "gatsby-plugin-manifest",
            options: {
                name: config.siteTitle,
                short_name: config.siteTitleAlt,
                description: config.siteDescription,
                start_url: config.pathPrefix,
                background_color: config.backgroundColor,
                theme_color: config.themeColor,
                display: "fullscreen",
                icon: "src/favicon.ico"
            }
        },
        "gatsby-plugin-offline",
        {
            resolve: "gatsby-plugin-google-analytics",
            options: {
                trackingId: "UA-41090209-2",
                head: false,
                anonymize: true,
                respectDNT: true
            }
        }
    ]
};
```
`package.json`: N/A 
`gatsby-node.js`: 
```
const path = require("path");
const _ = require("lodash");

exports.onCreateNode = ({ node, actions }) => {
  const { createNodeField } = actions;
  let slug;
  if (node.internal.type === "MarkdownRemark") {
    if (
      Object.prototype.hasOwnProperty.call(node, "frontmatter") &&
      Object.prototype.hasOwnProperty.call(node.frontmatter, "slug")
    ) {
      slug = `/${_.kebabCase(node.frontmatter.slug)}`;
    }
    if (
      Object.prototype.hasOwnProperty.call(node, "frontmatter") &&
      Object.prototype.hasOwnProperty.call(node.frontmatter, "title")
    ) {
      slug = `/${_.kebabCase(node.frontmatter.title)}`;
    }
    createNodeField({ node, name: "slug", value: slug });
  }
};

exports.createPages = ({ graphql, actions }) => {
  const { createPage } = actions;

  return new Promise((resolve, reject) => {
    const postPage = path.resolve("src/templates/post.js");
    const categoryPage = path.resolve("src/templates/category.js");
    resolve(
      graphql(`
        {
          posts: allMarkdownRemark {
            edges {
              node {
                fields {
                  slug
                }
                frontmatter {
                  title
                  category
                  path
                }
              }
            }
          }
        }
      `).then(result => {
        if (result.errors) {
          console.log(result.errors);
          reject(result.errors);
        }

        const posts = result.data.posts.edges;

        posts.forEach((edge, index) => {
          const next = index === 0 ? null : posts[index - 1].node;
          const prev =
            index === posts.length - 1 ? null : posts[index + 1].node;

          createPage({
            path: edge.node.frontmatter.path,
            component: postPage,
            context: {
              slug: edge.node.fields.slug,
              prev,
              next
            }
          });
        });

        let categories = [];

        _.each(posts, edge => {
          if (_.get(edge, "node.frontmatter.category")) {
            categories = categories.concat(edge.node.frontmatter.category);
          }
        });

        categories = _.uniq(categories);

        categories.forEach(category => {
          createPage({
            path: `/categories/${_.kebabCase(category)}`,
            component: categoryPage,
            context: {
              category
            }
          });
        });
      })
    );
  });
};
```
`gatsby-browser.js`: N/A 
`gatsby-ssr.js`: N/A

**Comment by pieh** [comment]
Sorry for comment spam :) Just noticed You use `icon: "src/favicon.ico"` and `.ico` isn't supported by `sharp` which is used to generate different sizes of icons - try using `.png`.

And we probably do need better validation (not only check if file exists but also if it's of supported format) + some error handling to at least notify where error was thrown

**Comment by thitemple** [comment]
That was it, `favicon.ico`. Thanks!

**Comment by MrUltimate** [comment]
I'm suddenly getting this error today, out of the blue. Using a normal png file. Did something change in Gatsby recent that might be breaking this?

**Comment by jackieluc** [comment]
@MrUltimiate This happened to me recently as well (after I was updating package dependencies). I fixed it with the following steps:

1. `rm -rf node_modules/`
2. `rm package-lock.json`
3. Update any of the following packages (ie. `npm install gatsby-plugin-sharp`) that you are using in your project. The list of plugins depend on the `sharp` package and was provided by the troubleshooting section of the [gatsby-plugin-manifest docs](https://www.gatsbyjs.org/packages/gatsby-plugin-manifest/#incompatible-library-version-sharpnode-requires-version-x-or-later-but-z-provides-version-y).

- gatsby-plugin-sharp
- gatsby-plugin-manifest
- gatsby-remark-images-contentful
- gatsby-source-contentful
- gatsby-transformer-sharp
- gatsby-transformer-sqip

4. Lastly, install the rest of your dependencies again with `npm install`

**Comment by MrUltimate** [comment]
Thanks, @jackieluc. I was actually able to fix it by simple running `npm upgrade` on the project.

**Comment by HemalR** [comment]
> 2\. rm package-lock.json

m -rf

**Comment by tplive** [comment]
@MrUltimate thanks! I was able to fix the problem by using npm upgrade too. Strangely, it didn't surface during local build, but was discovered in the Netlify build pipeline.

**Comment by jcarroll-ceo** [comment]
Understood this is CLOSED - but for the record (and future 'me's' that come here looking for a solution.... please note the following:

For me, this occurred after I installed gatsby's manifest plugin. I tried all of the solutions above (removing node_modules folder, reinstalling, audit fix, upgrading, etc etc...). 

For the record, it would build locally just fine... but once deployed to Netlify (even after a 'clear cache') it would still break.  

The ONLY solution that worked for me was adding: 

"resolutions": {
    "sharp": "0.24.0"
  },

in my package.json.

Good luck!

**Comment by thatgriffith** [comment]
@jackieluc thanks, finally solved my problem after 2 days.

I am using Gatsby with Netlify and its CI/CD. All of the sudden when a new build failed with same error as @thitemple. I tried a lot of different solution but finally this worked for me:
`npm install gatsby-plugin-sharp@latest gatsby@latest gatsby-tranformer-sharp@latest  gatsby-plugin-manifest@latest`

**Comment by Inambe** [comment]
I was facing a similar problem. In my case: I was using `gatsby-plugin-favicon` which I think is now deprecated (there is no announcement or anything though). I just had to remove `gatsby-plugin-favicon` and use `gatsby-plugin-manifest` which gets updated regularly and the site works fine now. also, `gatsby-plugin-favicon` is very easily replaceable by `gatsby-plugin-manifest`, I just had to change few lines.
